wstakemgr/db.go

```diff
@@ -1153,6 +1153,48 @@ func duplicateExistsInInvalTickets(record *chainhash.Hash,
 	return false
 }
 
+// removeStakePoolInvalUserTickets removes the ticket hash from the inval
+// ticket bucket.
+func removeStakePoolInvalUserTickets(ns walletdb.ReadWriteBucket, scriptHash [20]byte,
+	record *chainhash.Hash) error {
+	oldRecords, _ := fetchStakePoolUserInvalTickets(ns, scriptHash)
+
+	// Don't need to remove records that don't exist.
+	if !duplicateExistsInInvalTickets(record, oldRecords) {
+		return nil
+	}
+
+	var newRecords []*chainhash.Hash
+	for i := range oldRecords {
+		if record.IsEqual(oldRecords[i]) {
+			newRecords = make([]*chainhash.Hash, len(oldRecords)-1)
+			newRecords = append(oldRecords[0:i], oldRecords[i+1:]...)
```

Comment:
probably faster to use copy directly, but internally it might be doing the same thing
copy(newRecords[:], oldRecords[0:i])
copy(newRecords[i:], oldRecords[i+1:])